getting-started/web3-contract.md

```diff
@@ -0,0 +1,336 @@
+---
+title: Using Web3 for Contracts
+description: Learn how to deploy Solidity-based contracts on Moonbeam with a simple script using web3.
+---
+
+#Setting Using Web3 to deploy smart contracts on Moonbeam ##Introduction  
+This guide walks you through the process of using the solidity compiler and Web3 to deploy and interact with a Solidity-based smart contract on a Moonbeam dev node. Given Moonbeam’s Ethereum compatibility features, the Web3 library can be used directly with a Moonbeam node.
+
+The examples on this guide are based on a Ubuntu 18.04 environment and assume that you have a local Moonbeam node running in --dev mode, you can find instructions to set up a local Moonbeam node [here](/getting-started/setting-up-a-node/).
+
+!!! note 
+    This tutorial was created using the pre-alpha release of [Moonbeam](https://github.com/PureStake/moonbeam/tree/moonbeam-tutorials). The Moonbeam platform, and the [Frontier](https://github.com/paritytech/frontier) components it relies on for Substrate-based Ethereum compatibility, are still under very active development. We have created this tutorial so you can test out Moonbeam’s Ethereum compatibility features. Even though we are still in development, we believe it’s important that interested community members and developers have the opportunity to start to try things with Moonbeam and provide feedback.
+
+##Checking Prerequisites 
+
+If you followed setting up a local Moonbeam node tutorial, you should have a local Moonbeam node producing blocks that looks like this:
+
+![Moonbeam local node](/images/web3-contract-1.png)
+
+In addition, for this tutorial, we need to install Node.js and the npm package manager. You can do this by running in your terminal:
+
+```
+sudo apt install nodejs
+sudo apt install npm
+```
+
+We can verify if everything installed correctly by querying the version for each package:
+
+```
+node -v
+npm -v
+```
+
+As of the writing of this guide, versions used were 8.10.0 and 3.5.2, respectively. Next, we can create a directory to store all our relevant files (in a separate path from the local Moonbeam node files), and create a simple package.json file by running:
+
+```
+mkdir incrementer
+cd incrementer/
+npm init --yes
+```
+
+With the package.json file created, we can then install both the Web3 and the Solidity compiler packages, by executing:
+
+```
+npm install --save web3

```

Comment:
If you find it more meaningful, you can also specify the versions in the command, ex:
`npm install --save web3@1.2.9 solc@0.6.10`

Reply:
Thanks for the tip